samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -75,7 +75,9 @@ StreamSpec getStreamSpec() {
     StreamSpec spec = (partitions == PARTITIONS_UNKNOWN) ?
         streamSpec : streamSpec.copyWithPartitionCount(partitions);
 
-    if (isIntermediate) {
+    // Append unique id to the batch intermediate streams
+    // Check the physical stream name is already generated first
+    if (isIntermediate && spec.getId().equals(spec.getPhysicalName())) {
```

Comment:
Could there be a case where the Id and the physical name are not set to the same value for batch? In either case, we should add a comment about how we can guarantee this block is only for batch.